documentation/reference/search-definitions-reference.html

```diff
@@ -2400,67 +2466,105 @@ <h2 id="field_types">Field types</h2>
+First, create a <a href="#struct">struct type</a> inside the document definition,
+then declare the struct field in a document or struct using the struct type name as the field type.
+Restrictions:
+<ul>
+  <li>Structs can <strong>only</strong> be searched in <a href="../streaming-search.html">streaming search</a> mode,
+      <strong>not</strong> in indexed search.</li>
+  <li>Structs can be fed, retrieved and used in document summaries in both indexed and streaming search.</li>
+</ul>
+
+Example:
 <pre>
-field foo type address {}
+struct person {
+  field first_name type string {}
+  field last_name type string {}
+}
+field my_person type person {
+  indexing: summary
+}
 </pre>
 <table class="table">
   <thead></thead><tbody>
     <tr>
       <th style="width:100px">Indexing</th>
-      <td>Structs have the same <a href="#type:map">indexing restrictions</a> as maps.</td>
+      <td>Not supported.</td>
     </tr><tr>
       <th>Attribute</th>
-      <td>Each field may or may not be an attribute according to its configuration.</td>
+      <td>Not supported.</td>
     </tr><tr>
       <th>Summary</th>
-      <td>Each field in the struct has its own summary configuration</td>
+      <td>Added as a struct. In streaming search each field in the struct can have its own summary configuration.</td>
     </tr>
   </tbody>
 </table>
 </td></tr>
 
 <tr><th id="type:map">map&lt;key-type,value-type&gt;</th>
 <td>
-Use to create a map of values. Use any other Vespa type as both key and value.
-These configurations are identical:
+Use to create a map where each unique key is mapped to a single value.
+Any other Vespa type is used as <em>key-type</em> and <em>value-type</em>.
+A map is handled as a struct with a <em>key</em> and <em>value</em> field with <em>key-type</em> and <em>value-type</em> as types.
+Restrictions:
+<ul>
+  <li>All map types can be searched in <a href="../streaming-search.html">streaming search</a> mode.</li>
+  <li>Some map types can be searched in indexed search. See table below for supported cases.</li>
+  <li>All map types can be fed, retrieved and used in document summaries in both indexed and streaming search.</li>
+</ul>
+
+Example:
 <pre>
-field mymap type map&lt;string, string&gt; {
+struct person {
+  field first_name type string {}
+  field last_name type string {}
 }
-
+field identities type map&lt;string, person&gt; {
+  indexing: summary
+  struct-field key { indexing: attribute }
+  struct-field value.last_name { indexing: attribute }
+}
+</pre>
+The entire <em>identities</em> field is part of document summary,
+and the struct fields <em>key</em> and <em>value.last_name</em> are attributes available for searching using the
+<a href="../query-language.html#same-element">sameElement</a> operator.
+The next example shows a map of primitive types, where the <em>key</em> and <em>value</em> struct fields are specified as attributes:
```

Comment:
Consider adding a paragraph break before this sentence since it's a slight change of context